src/browser/index.ts

```diff
@@ -0,0 +1,156 @@
+import { CallbackType } from '../auth/enums';
+import { parseRelyingPartyId, getClientDataJson } from '../util/webauthn';
+import FRStep from 'fr-auth/fr-step';
+import { WebAuthnAuthenticationMetadata, WebAuthnRegistrationMetadata } from 'browser/interfaces';
+
+abstract class WebAuthn {
+  public static async register(step: FRStep) {
+    const { hiddenCallback, metadataCallback } = WebAuthn.getCallbacks(step);
+    const publicKey = WebAuthn.createRegistrationPublicKey(metadataCallback.getOutputValue());
+    const credentials = await WebAuthn.getRegistrationCredentials(publicKey);
+    hiddenCallback.setInputValue(credentials);
+    return step;
+  }
+
+  public static async authenticate(step: FRStep) {
+    const { hiddenCallback, metadataCallback } = WebAuthn.getCallbacks(step);
+    const publicKey = WebAuthn.createAuthenticationPublicKey(metadataCallback.getOutputValue());
+    const credentials = await WebAuthn.getAuthenticationCredentials(publicKey);
+    hiddenCallback.setInputValue(credentials);
+    return step;
+  }
+
+  public static getCallbacks(step: FRStep) {
+    const hiddenCallback = step.getCallbackOfType(CallbackType.HiddenValueCallback);
+    const metadataCallback = step.getCallbackOfType(CallbackType.MetadataCallback);
+    return { hiddenCallback, metadataCallback };
+  }
+
+  public static async getAuthenticationCredentials(publicKey: PublicKeyCredentialRequestOptions) {
```

Comment:
For the auth trees to work correctly, we need to handle more possible outcomes.  The `outcome` value maps to connection points on the auth tree node and determines branching logic.
```
    try {
      if (!window.PublicKeyCredential) {
        return 'unsupported';
      } else {
        // do the needful
        return `${clientDataJSON}::${attestationObject}::${credential.id}`;
      }
    } catch (err) {
      if (err.name === 'NotAllowedError') {
        return 'ERROR::NotAllowedError:The operation either timed out or was not allowed';
      }
      return `ERROR::${err}`;
    }
```